include/common_structs.h

```diff
@@ -247,12 +252,12 @@ typedef struct ScriptInstance {
     /* 0x0F0 */ s32 loopCounterTable[8];
     /* 0x110 */ u8 switchBlockState[8];
     /* 0x118 */ s32 switchBlockValue[8];
-    /* 0x138 */ s32* buffer;
-    /* 0x13C */ UNK_PTR array;
-    /* 0x140 */ UNK_PTR flagArray;
+    /* 0x138 */ ScriptBufferValue* buffer;
```

Comment:
I'm not sure I really like naming a struct Thing**Value** but I don't have a better name atm..hm

Reply:
I wonder if I could declare _buffer_ as a union between a pointer to an int or a pointer to a float, rather than the type of what it points to. That is:
```diff
- buffer[0].s
+ buffer.s[0]
```

Reply:
If it helps, Star Rod uses the types `IntBuffer` and `FloatBuffer` for the structs you pass to the `SetBuffer` and `SetFloatBuffer` script commands.

Reply:
okay, yeah I can't think of a better name so moop